controllers/azuremachine_reconciler.go

```diff
@@ -51,6 +51,23 @@ func newAzureMachineService(machineScope *scope.MachineScope) (*azureMachineServ
 		return nil, errors.Wrap(err, "failed creating a NewCache")
 	}
 
+	// If ExtendedLocation exists, remove Availability set service.
+	if machineScope.ExtendedLocation() != nil {
```

Comment:
we shouldn't need to do this, we can simply make the availability set service reconcile a no-op when `ExtendedLocation` is enabled, by returning a nil spec in machine.go. This should already be the case with https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2836/files#diff-b6931ac61f37a13f66fc7836e039bf714a0553d5db4f313cc8ef31ca5f5647b9R498

We can't remove services that are unused based on every Machine configuration that doesn't need them in here otherwise the code would become a mess of if/else statements.

Reply:
Yes, we don't need the if statement as nil spec in machine.go will help return nil in availability set reconcile func.

Thanks for your kind and helpful comment.